### PR TITLE
[TR] PIM-3940 Rework js managers

### PR DESCRIPTION
**Pull request by filipsalpe**

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/form/config-provider.js [hunk 1]
```diff
@@ -6,38 +6,23 @@ define(
...
         };
 
         return {
             getExtensionMap: function () {
-                var promise = $.Deferred();
-
-                loadConfig().done(function (config) {
-                    promise.resolve(config.extensions);
+                return loadConfig().then(function (config) {
+                    return config.extensions;
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/form/config-provider.js]: weird way of doing that :/

Comment (filipsalpe) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/form/config-provider.js]: why?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/manager/entity-manager.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+'use strict';
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/manager/entity-manager.js]: entity manager ? Maybe object manager ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/repository/entity-repository.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            } else {
+                entityListPromise = null;
+            }
+        };
+
+        return this;
+    };
+
+    // Provide a Backbone-like extension point
+    EntityRepository.extend = Backbone.Model.extend;
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/repository/entity-repository.js]: wow, very nice

**Comment by sumbobyboys** [comment]
GTM

**Pull request merged**